src/index.tsx

```diff
@@ -1,280 +1,142 @@
-        type: DispatchType.Transition,
-        next,
-      })
-    );
+interface SetContextEvent
+  { type: "SET_CONTEXT"
+  , updater: Machine.ContextUpdater.Impl
+  }
 
-    // The updater function sends an internal event to the reducer to trigger the actual update
-    const setContext = (updater?: ContextUpdater<Context>) => {
-      if (updater) {
-        dispatch({
-          type: DispatchType.Update,
-          updater,
-        });
-      }
-      return { send };
-    };
+interface SendEvent
+  { type: "SEND"
+  , sendable: Machine.Sendable.Impl
+  }
 
-    useEffect(() => {
-      const exit = config.states[machine.value]?.effect?.({
-        send,
-        setContext,
-        event: machine.event,
-        context: machine.context,
+type InternalEvent =
+  | SetContextEvent
+  | SendEvent
+
+const createLogger = (definition: Machine.Definition.Impl) =>
+  (groupLabel: string, ...nested: [string, any][]) => {
+    if (!definition.verbose) return;
+    if (process.env.NODE_ENV === "development") {
+      console.groupCollapsed("%cuseStateMachine", "color: #888; font-weight: lighter;", groupLabel);
+      nested.forEach(message => {
+        console.log(message[0], message[1]);
       });
-      return typeof exit === 'function'
-        ? () => exit({ send, setContext, event: machine.event, context: machine.context })
-        : undefined;
-      // We are bypassing the linter here because we deliberately want the effects to run:
-      // - When the machine state changes or
-      // - When a different event was sent (e.g. self-transition)
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [machine.value, machine.event]);
-
-    return [machine, send];
-  };
-}
+      console.groupEnd();
+    }
+  }
 
-/**
- * A finite state machine is always on one given state, and reacts to events by transitioning to a different state and triggering effects.
- *
- * @version 1.0.0-beta.1
- * @see [github.com/cassiozen/useStateMachine](https://github.com/cassiozen/useStateMachine)
- */
-export default function useStateMachine(): UseStateMachineWithContext<undefined, undefined>;
-export default function useStateMachine<Context>(context: Context): UseStateMachineWithContext<Context, undefined>;
-export default function useStateMachine<Context, Events>(context: Context): UseStateMachineWithContext<Context, Events>;
-export default function useStateMachine<Context>(
-  context?: Context
-): UseStateMachineWithContext<Context | undefined, undefined>;
-export default function useStateMachine<Context, Events>(
-  context?: Context
-): UseStateMachineWithContext<Context | undefined, Events>;
-export default function useStateMachine<Context, Events>(
-  context?: Context
-): UseStateMachineWithContext<Context | undefined, Events> {
-  return useStateMachineImpl(context);
-}
+const useStateMachine = useStateMachineImpl as UseStateMachine
+export default useStateMachine;
+export const t = <T extends unknown>() => null as T
```

Comment:
I was thinking of making it 
```ts
export const $$t = Symbol("t");
export const t = <T extends unknown>() => $$t as T;
```
and then add a runtime check if `$$t` is what the users are passing in the schema or not, if not then log a warning that they are supposed to use `t`. But then it's trivial and would add to the size of the library.

Reply:
Just so we're clear: I don't think library size is the most important metric here. I used it as a catchphrase because it happened to be half a kb, but I don't mind it being bigger.

Reply:
Haha yeah I'm aware. But I'd only suggest to add this if you think I'd be useful, so I'll let you take the call.

Reply:
I do think it would be useful ;)

Reply:
We should add it then! I think a separate `checkSchema` would be good.

Reply:
Perhaps something like...
```ts
const checkSchema = schema => {
  if (!schema) return;
  checkContextSchema(schema.context);
  checkEventsSchema(schema.events);
}

const checkContextSchema = schema = {
  if (!definition.schema.context) return;
  if (defintion.schema.context === $$t) return;
  warn("whatever")
}

const checkEventsSchema = schema => {
  if (!definition.schema.events) return;
  if (typeof definition.schema.events !== "obejct") {
    warn("...")
    return;
  }
  if (
    R.entries(defintion.schema.events)
    .every(([k, v]) => k === "$$exhuastive" ? true : v === $$t)
  ) return;
  warn("whatever")
}
```

Reply:
Actually you know what I realized we don't need to do this I can enforce people use `t` via types itself. I'll update the types.